database.go

```diff
@@ -166,3 +164,71 @@ func (e ErrRequiredAttributeMissing) Error() string {
 		string(e),
 	)
 }
+
+type BinariesUsages map[int][]*BinaryReference
```

Comment:
Do we need `BinariesUsages` to be an exported type?

Reply:
No.